lib/fluent/plugin/filter_viaq_data_model.rb

```diff
@@ -104,12 +216,244 @@ def delempty(thing)
+      "_GID"               => "GID",
+      "_MACHINE_ID"        => "MACHINE_ID",
+      "_PID"               => "PID",
+      "_SELINUX_CONTEXT"   => "SELINUX_CONTEXT",
+      "_SYSTEMD_CGROUP"    => "SYSTEMD_CGROUP",
+      "_SYSTEMD_OWNER_UID" => "SYSTEMD_OWNER_UID",
+      "_SYSTEMD_SESSION"   => "SYSTEMD_SESSION",
+      "_SYSTEMD_SLICE"     => "SYSTEMD_SLICE",
+      "_SYSTEMD_UNIT"      => "SYSTEMD_UNIT",
+      "_SYSTEMD_USER_UNIT" => "SYSTEMD_USER_UNIT",
+      "_TRANSPORT"         => "TRANSPORT",
+      "_UID"               => "UID"
+    }
+
+    JOURNAL_FIELD_MAP_SYSTEMD_U = {
+      "CODE_FILE"         => "CODE_FILE",
+      "CODE_FUNCTION"     => "CODE_FUNCTION",
+      "CODE_LINE"         => "CODE_LINE",
+      "ERRNO"             => "ERRNO",
+      "MESSAGE_ID"        => "MESSAGE_ID",
+      "RESULT"            => "RESULT",
+      "UNIT"              => "UNIT",
+      "SYSLOG_FACILITY"   => "SYSLOG_FACILITY",
+      "SYSLOG_IDENTIFIER" => "SYSLOG_IDENTIFIER",
+      "SYSLOG_PID"        => "SYSLOG_PID"
+    }
+
+    JOURNAL_FIELD_MAP_SYSTEMD_K = {
+      "_KERNEL_DEVICE"    => "KERNEL_DEVICE",
+      "_KERNEL_SUBSYSTEM" => "KERNEL_SUBSYSTEM",
+      "_UDEV_SYSNAME"     => "UDEV_SYSNAME",
+      "_UDEV_DEVNODE"     => "UDEV_DEVNODE",
+      "_UDEV_DEVLINK"     => "UDEV_DEVLINK",
+    }
+
+    JOURNAL_TIME_FIELDS = ['_SOURCE_REALTIME_TIMESTAMP', '__REALTIME_TIMESTAMP']
+
+    def process_journal_fields(tag, time, record, fmtr_type)
+      systemd_t = {}
+      JOURNAL_FIELD_MAP_SYSTEMD_T.each do |jkey, key|
+        if record[jkey]
+          systemd_t[key] = record[jkey]
+        end
+      end
+      systemd_u = {}
+      JOURNAL_FIELD_MAP_SYSTEMD_U.each do |jkey, key|
+        if record[jkey]
+          systemd_u[key] = record[jkey]
+        end
+      end
+      systemd_k = {}
+      JOURNAL_FIELD_MAP_SYSTEMD_K.each do |jkey, key|
+        if record[jkey]
+          systemd_k[key] = record[jkey]
+        end
+      end
+      unless systemd_t.empty?
+        (record['systemd'] ||= {})['t'] = systemd_t
+      end
+      unless systemd_u.empty?
+        (record['systemd'] ||= {})['u'] = systemd_u
+      end
+      unless systemd_k.empty?
+        (record['systemd'] ||= {})['k'] = systemd_k
+      end
+      begin
+        pri_index = ('%d' % record['PRIORITY'] || 9).to_i
+        if pri_index < 0
+          pri_index = 9
+        elsif pri_index > 9
+          pri_index = 9
+        end
+      rescue
+        pri_index = 9
+      end
+      record['level'] = ["emerg", "alert", "crit", "err", "warning", "notice", "info", "debug", "trace", "unknown"][pri_index]
+      JOURNAL_TIME_FIELDS.each do |field|
+        if record[field]
+          record['time'] = Time.at(record[field].to_f / 1000000.0).utc.to_datetime.rfc3339(6)
+          break
+        end
+      end
+      if fmtr_type == :sys_journal
+        record['message'] = record['MESSAGE']
+        if record['_HOSTNAME'].eql?('localhost') && @docker_hostname
```

Comment:
Do we need to invert this incase the recored does not have hostname.. `'localhost'.eql?(record['_HOSTNAME'])`

Reply:
no - if no `_HOSTNAME` then `record['_HOSTNAME']` is `nil` and `nil.eql?('localhost')` is `false`